In [18]:
import pandas as pd


In [19]:
df = pd.read_csv('loan.csv',low_memory=False)

In [20]:
df.shape

(39717, 111)

## Data cleaning

#### Lets understand the Null Values present

In [21]:
df.isna().sum()

id                                0
member_id                         0
loan_amnt                         0
funded_amnt                       0
funded_amnt_inv                   0
                              ...  
tax_liens                        39
tot_hi_cred_lim               39717
total_bal_ex_mort             39717
total_bc_limit                39717
total_il_high_credit_limit    39717
Length: 111, dtype: int64

#### There are total 39717 values in the data frame but some columns seems to have all the values as null values. <br> We should remove such columns as they won't contribute anything to the analysis

In [22]:
df = df.loc[:,df.isnull().sum() != len(df)]
df.shape

(39717, 57)

In [23]:
df.isna().sum()

id                                0
member_id                         0
loan_amnt                         0
funded_amnt                       0
funded_amnt_inv                   0
term                              0
int_rate                          0
installment                       0
grade                             0
sub_grade                         0
emp_title                      2459
emp_length                     1075
home_ownership                    0
annual_inc                        0
verification_status               0
issue_d                           0
loan_status                       0
pymnt_plan                        0
url                               0
desc                          12942
purpose                           0
title                            11
zip_code                          0
addr_state                        0
dti                               0
delinq_2yrs                       0
earliest_cr_line                  0
inq_last_6mths              

#### Now we are left with 57 columns. Lets drop all the columns which contains very high number of null values and seem insignificant for our analysis

In [24]:
col_to_drop = ['desc','mths_since_last_delinq','mths_since_last_record','next_pymnt_d']
df = df.drop(col_to_drop,axis=1)
df.shape

(39717, 53)

#### So now we are left with 53 columns. Lets now see which columns have just a single value and we can remove such columns since it won't contribute much to analysis